src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/RoslynCompletionSet.cs

```diff
@@ -12,16 +11,15 @@
 using Microsoft.VisualStudio.Language.Intellisense;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Editor;
-using Roslyn.Utilities;
 using VSCompletion = Microsoft.VisualStudio.Language.Intellisense.Completion;
 using Microsoft.CodeAnalysis.Snippets;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Presentation
 {
-#if NEWCOMPLETION
-    internal sealed class CompletionSet3 : CompletionSet2
+#if DEV15
```

Comment:
I still contend this can be cleaner with a pseudo mixin pattern.

Here's how it would work:

We'd have internal types (call them Foo for now) that didn't inherit from the editor types.  These would represent our code implementation of the completion set code for Dev14, and would be subclassed by our Dev15 impl (which we'll call 'Bar').

Then, we'd have two actual roslyn CompletionSet types for Dev14 and Dev15 which appropriately suclasses CompletionSet or CompletionSet2.  It would only contain an instance of that Foo type.  The dev14 version would instatiate a Foo and delegate everything to it.  The dev15 version would instantiate a Bar and delegate everything to it.

Yes, this would be ugly in the sense that we'd have two classes that did nothing but forward along calls to the actual impl type they held onto.  Lacking real mixins, that's the best we could do.

However, it would mean:
1. no ifdefs.
2. Actual appropriate subclassing of the part of the API we control.